src/test/java/ai/starlake/transpiler/JSQLColumnResolverTest.java

```diff
@@ -73,6 +73,35 @@ void testSimpleSchemaProvider() throws JSQLParserException, SQLException {
         .isEqualTo(new String[] {res.getTableName(6), res.getColumnName(6)});
   }
 
+  @Test
+  void testWithWrongTable() {
+    JdbcMetaData metaData = new JdbcMetaData("", "")
+        .addTable("a", new JdbcColumn("col1"), new JdbcColumn("col2"), new JdbcColumn("col3"))
+        .addTable("b", new JdbcColumn("col1"), new JdbcColumn("col2"), new JdbcColumn("col3"));
+
+    //should be exception because bb table is not exist
+    org.junit.jupiter.api.Assertions.assertThrows(JSQLParserException.class, () -> JSQLColumResolver.getResultSetMetaData("select * from a where a.col1 in (select bb.col1 from bb)", metaData));
```

Comment:
`JSQLParserException` is very generic and can mean anything.
If you want to make this test meaniningful then please check if Table `bb` is pointed out and blamed correctly.